File: batch/src/test/java/com/example/batch/fixture/InitProcessor.java
package com.example.batch.fixture;

import com.example.batch.entity.student.Teacher;
import com.example.batch.entity.student.TeacherRepository;
import org.springframework.stereotype.Component;
import org.springframework.transaction.annotation.Transactional;

@Component
public class InitProcessor {
    private final TeacherRepository teacherRepository;

    public InitProcessor(TeacherRepository teacherRepository) {
        this.teacherRepository = teacherRepository;
    }

    @Transactional
    public void saveTeacher() {
        for (int i = 0; i < 30; i++) {
            teacherRepository.save(Teacher.builder()
                    .name("name" + i)
                    .subject("subject" + i)
                    .build()
            );
        }
    }
}

File: jdsl-example/src/main/kotlin/com/example/jdslexample/common/init/InitProcessor.kt
package com.example.jdslexample.common.init

import com.example.jdslexample.persistence.entity.MemberEntity
import com.example.jdslexample.persistence.entity.RoleEntity
import com.example.jdslexample.persistence.repository.MemberRepository
import com.example.jdslexample.persistence.repository.RoleRepository
import org.springframework.stereotype.Component
import org.springframework.transaction.annotation.Transactional

@Component
class InitProcessor(
    private val testMemberInit: TestMemberInit
) {
    //    @PostConstruct
    fun init() {
        testMemberInit.initAll()
    }
}

@Component
@Transactional
class TestMemberInit(
    private val memberRepository: MemberRepository,
    private val roleRepository: RoleRepository,
) {
    lateinit var roleUser: RoleEntity
    lateinit var roleAdmin: RoleEntity

    @Transactional
    fun initAll() {
        roleInit()
        memberInit()
    }

    fun roleInit() {
        roleUser = RoleEntity(name = "user")
        roleAdmin = RoleEntity(name = "admin")
        roleRepository.save(roleUser)
        roleRepository.save(roleAdmin)
    }

    fun memberInit() {
        val memberEntity = MemberEntity(name = "test1")
        memberEntity.addRole(roleUser)
        memberRepository.save(memberEntity)

        val adminMember = MemberEntity(name = "test2")
        adminMember.addRole(roleAdmin)
        memberRepository.save(adminMember)

        val allMember = MemberEntity(name = "test3")
        allMember.addRole(roleAdmin)
        allMember.addRole(roleAdmin)
        memberRepository.save(allMember)

    }
}